# ***Piyush Anil Patil***

**The objective is to furnish the San Francisco Chronicle with a report that elucidates the principal findings pertaining to income dynamics in the city, which significantly impact its cost of living.**

#1.***Objective of the Proposed Research***


The primary goal of this research is to explore the extensive dataset of SF Employee Compensation spanning over a year from 2013 to 2020. The dataset contains 683278 Rows and 22 Columns. Each Data set representing crucial information such as Organization, Job Family, Job, Year, Organization, Department, Salaries, Overtime, Retirement, Health and Dental Benefit, Total Benefits and Total Compensation


1. **Provide a Comprehensive Overview**: The primary objective is to offer readers a comprehensive overview of the income landscape in San Francisco. This should include key statistics such as median income levels, income distribution, and trends over recent years.

2. **Highlight Income Disparities**: To shed light on income inequality within the city, one objective is to highlight disparities in income, especially between different sectors of the population, such as tech industry workers and service workers.

3. **Explore the Impact of the Various Job Sector**: Another objective is to delve into the influence of the Job sector on income dynamics. This includes examining the extent to which tech industry jobs contribute to high median incomes and exploring any potential consequences for other residents.

4. **Examine Income vs. Cost of Living**: To provide a clearer understanding of the challenges faced by San Francisco residents, the report should aim to examine the relationship between income levels and the cost of living in the city, particularly focusing on the affordability of housing.

#***2.DATA EXPLORATION***

In [6]:
# Load the dataset
sf.data <- read.csv("//employee-compensation.csv")

 2.1. Calculate Columns and Rows of the Data Set

In [7]:
sf.data <- sf.data[sf.data$Year.Type == "Fiscal",]
head(sf.data,5)

Organization.Group.Code Job.Family.Code Job.Code Year.Type Year
298165 1                       1000            1021     Fiscal    2013
298166 1                       1000            1023     Fiscal    2013
298167 1                       1000            1031     Fiscal    2013
298168 1                       1000            1054     Fiscal    2013
298169 1                       1000            1070     Fiscal    2013
       Organization.Group Department.Code Department          Union.Code
298165 Public Protection  ADP             ADP Adult Probation 21        
298166 Public Protection  ADP             ADP Adult Probation 21        
298167 Public Protection  ADP             ADP Adult Probation 21        
298168 Public Protection  ADP             ADP Adult Probation 21        
298169 Public Protection  ADP             ADP Adult Probation 21        
       Union                                           ⋯ Employee.Identifier
298165 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37730              
298166 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 26574              
298167 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯  8148              
298168 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 27436              
298169 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 51426              
       Salaries  Overtime Other.Salaries Total.Salary Retirement
298165  58368.56 0        0               58368.56    10628.94  
298166  99944.00 0        0               99944.00    17700.03  
298167  59585.91 0        0               59585.91    10850.65  
298168  65100.00 0        0               65100.00    11529.18  
298169 134251.77 0        0              134251.77    23775.87  
       Health.and.Dental Other.Benefits Total.Benefits Total.Compensation
298165 12801.79          5254.13        28684.86        87053.42         
298166 12801.79          8311.72        38813.54       138757.54         
298167 12801.79          4461.25        28113.69        87699.60         
298168  8616.59          5139.40        25285.17        90385.17         
298169 12801.79          9733.22        46310.88       180562.65

2.2 Summary of the Data

In [9]:
summary(sf.data)


 Organization.Group.Code Job.Family.Code      Job.Code        
 Min.   :1.000           Length:385113      Length:385113     
 1st Qu.:2.000           Class :character   Class :character  
 Median :2.000           Mode  :character   Mode  :character  
 Mean   :2.981                                                
 3rd Qu.:4.000                                                
 Max.   :7.000                                                
                                                              
  Year.Type              Year      Organization.Group Department.Code   
 Length:385113      Min.   :2013   Length:385113      Length:385113     
 Class :character   1st Qu.:2015   Class :character   Class :character  
 Mode  :character   Median :2017   Mode  :character   Mode  :character  
                    Mean   :2017                                        
                    3rd Qu.:2018                                        
                    Max.   :2020                          

2.3. First 5 Rows of the Data

In [10]:
head(sf.data)

Organization.Group.Code Job.Family.Code Job.Code Year.Type Year
298165 1                       1000            1021     Fiscal    2013
298166 1                       1000            1023     Fiscal    2013
298167 1                       1000            1031     Fiscal    2013
298168 1                       1000            1054     Fiscal    2013
298169 1                       1000            1070     Fiscal    2013
298170 1                       1200            1232     Fiscal    2013
       Organization.Group Department.Code Department          Union.Code
298165 Public Protection  ADP             ADP Adult Probation 21        
298166 Public Protection  ADP             ADP Adult Probation 21        
298167 Public Protection  ADP             ADP Adult Probation 21        
298168 Public Protection  ADP             ADP Adult Probation 21        
298169 Public Protection  ADP             ADP Adult Probation 21        
298170 Public Protection  ADP             ADP Adult Probation 21        
       Union                                           ⋯ Employee.Identifier
298165 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37730              
298166 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 26574              
298167 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯  8148              
298168 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 27436              
298169 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 51426              
298170 Prof & Tech Engineers - Miscellaneous, Local 21 ⋯ 37983              
       Salaries  Overtime Other.Salaries Total.Salary Retirement
298165  58368.56 0        0               58368.56    10628.94  
298166  99944.00 0        0               99944.00    17700.03  
298167  59585.91 0        0               59585.91    10850.65  
298168  65100.00 0        0               65100.00    11529.18  
298169 134251.77 0        0              134251.77    23775.87  
298170  79407.01 0        0               79407.01    14460.10  
       Health.and.Dental Other.Benefits Total.Benefits Total.Compensation
298165 12801.79          5254.13        28684.86        87053.42         
298166 12801.79          8311.72        38813.54       138757.54         
298167 12801.79          4461.25        28113.69        87699.60         
298168  8616.59          5139.40        25285.17        90385.17         
298169 12801.79          9733.22        46310.88       180562.65         
298170 12801.79          6484.23        33746.12       113153.13

2.4. Columns names of the Dataset

In [11]:
colnames(sf.data)

[1] "Organization.Group.Code" "Job.Family.Code"        
 [3] "Job.Code"                "Year.Type"              
 [5] "Year"                    "Organization.Group"     
 [7] "Department.Code"         "Department"             
 [9] "Union.Code"              "Union"                  
[11] "Job.Family"              "Job"                    
[13] "Employee.Identifier"     "Salaries"               
[15] "Overtime"                "Other.Salaries"         
[17] "Total.Salary"            "Retirement"             
[19] "Health.and.Dental"       "Other.Benefits"         
[21] "Total.Benefits"          "Total.Compensation"

2.5. Last 5 Rows of the Data Set

In [12]:
tail(sf.data)

Organization.Group.Code Job.Family.Code Job.Code Year.Type Year
683272 6                       4300            4334     Fiscal    2020
683273 6                       4300            4334     Fiscal    2020
683274 6                       4300            4334     Fiscal    2020
683275 6                       4300            4334     Fiscal    2020
683276 6                       4300            4337     Fiscal    2020
683277 6                       4300            4337     Fiscal    2020
       Organization.Group               Department.Code Department             
683272 General Administration & Finance TTX             Treasurer/Tax Collector
683273 General Administration & Finance TTX             Treasurer/Tax Collector
683274 General Administration & Finance TTX             Treasurer/Tax Collector
683275 General Administration & Finance TTX             Treasurer/Tax Collector
683276 General Administration & Finance TTX             Treasurer/Tax Collector
683277 General Administration & Finance TTX             Treasurer/Tax Collector
       Union.Code Union                  ⋯ Employee.Identifier Salaries 
683272 790        SEIU, Local 1021, Misc ⋯ 42444671            105738.31
683273 790        SEIU, Local 1021, Misc ⋯ 43836214             98444.94
683274 790        SEIU, Local 1021, Misc ⋯ 43966403              9416.71
683275 790        SEIU, Local 1021, Misc ⋯ 44016180             96475.00
683276 790        SEIU, Local 1021, Misc ⋯ 42423439            115943.00
683277 790        SEIU, Local 1021, Misc ⋯ 42439576            115943.00
       Overtime Other.Salaries Total.Salary Retirement Health.and.Dental
683272 303.6      0            106041.91    22038.95   15451.43         
683273   0.0      0             98444.94    20540.24   15275.18         
683274   0.0      0              9416.71     1904.43    1468.77         
683275   0.0    576             97051.00    20241.56   14100.16         
683276   0.0      0            115943.00    24188.37   15275.18         
683277 504.9      0            116447.90    24188.37   15275.18         
       Other.Benefits Total.Benefits Total.Compensation
683272 8406.34        45896.72       151938.63         
683273 8019.49        43834.91       142279.85         
683274  743.51         4116.71        13533.42         
683275 7897.00        42238.72       139289.72         
683276 9018.34        48481.89       164424.89         
683277 9516.61        48980.16       165428.06

Adding library dplyr
Adding library tidyverse
2.6. Getting the Glimpse of the Dataset

In [13]:
library(tidyverse)

library(dplyr)
glimpse(sf.data)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


Rows: 385,113
Columns: 22
$ Organization.Group.Code <int> 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1…
$ Job.Family.Code         <chr> "1000", "1000", "1000", "1000", "1000", "1200"…
$ Job.Code                <chr> "1021", "1023", "1031", "1054", "1070", "1232"…
$ Year.Type               <chr> "Fiscal", "Fiscal", "Fiscal", "Fiscal", "Fisca…
$ Year                    <int> 2013, 2013, 2013, 2013, 2013, 2013, 2013, 2013…
$ Organization.Group      <chr> "Public Protection", "Public Protection", "Pub…
$ Department.Code         <chr> "ADP", "ADP", "ADP", "ADP", "ADP", "ADP", "ADP…
$ Department              <chr> "ADP Adult Probation", "ADP Adult Probation", …
$ Union.Code              <int> 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21, 21…
$ Union                   <chr> "Prof & Tech Engineers - Miscellaneous, Local …
$ Job.Family              <chr> "Information Systems", "Information Systems", …
$ Job                     <chr> "IS Administrator 1", "IS Administrator 3", "I…
$ Employee.Ide

#3.*Subsets of the Dataset*

The following Subset contains information about Salaries,Overtime,Retirement,Health.and.Dental which will help in analyzing condition in SF

3.1. Calculating top 10 the highest salary for year 2020

In [174]:
#Counting highest salary
sf_salary <- sf.data[c('Year', 'Job','Salaries')]
sf_salary_1 <- sf_salary %>% group_by('Job') %>% arrange(desc(Salaries))
sf_salary_1 <- sf_salary_1[sf_salary_1$Year == "2020",]%>% head(n=10)
#sf_salary_1 %>% head(n=10)

sf_salary_1[4] <- NULL

sf_salary_1

Year Job                          Salaries
1  2020 Chief Investment Officer     584297.6
2  2020 Managing Director            488641.2
3  2020 Managing Director            488218.4
4  2020 Physician Administrator, DPH 465612.0
5  2020 Managing Director            465400.7
6  2020 Managing Director            454947.4
7  2020 Physician Administrator, DPH 443441.0
8  2020 Director                     401979.6
9  2020 Executive Contract Employee  390376.0
10 2020 Director                     379244.9

3.2. Calculating Highest Overtime of Different Job sector of year 2020

In [152]:
sf_overtime<- sf.data[c( 'Year','Job','Overtime')]
sf_overtime <- sf_overtime %>% group_by('Job') %>% arrange(desc(Overtime))
sf_overtime <- sf_overtime[sf_overtime$Year == "2020",]%>% head(n=10)
#sf_overtime %>% head(n=10)

sf_overtime[4] <- NULL

sf_overtime

Year Job                   Overtime
1  2020 Senior Deputy Sheriff 317589.4
2  2020 Deputy Sheriff        272856.4
3  2020 Deputy Sheriff        262417.2
4  2020 Senior Deputy Sheriff 249774.1
5  2020 Senior Deputy Sheriff 238600.4
6  2020 Sheriff's Lieutenant  214921.0
7  2020 Sheriff's Lieutenant  214900.2
8  2020 Deputy Sheriff        213483.0
9  2020 Senior Deputy Sheriff 213198.3
10 2020 Sheriff's Lieutenant  209889.9

3.3. Calculating Highest Health and Dental Benefits of Different Job sector of year 2020

In [70]:
sf_health <- sf.data[c( 'Year','Job','Health.and.Dental')]
sf_health <- sf_health %>% group_by('Job') %>% arrange(desc(Health.and.Dental))
sf_health <- sf_health[sf_health$Year == "2020",]%>% head(n=10)
#sf_health %>% head(n=10)

sf_health[4] <- NULL

sf_health

Year Job                        Health.and.Dental
1  2020 Manager VI                 49000.22         
2  2020 Manager III                43446.48         
3  2020 Court Manager              35633.00         
4  2020 Court Computer Sys Manager 35633.00         
5  2020 Manager I                  33480.25         
6  2020 Manager II, MTA            33479.10         
7  2020 Manager V                  32363.01         
8  2020 Lieutenant 3               32257.85         
9  2020 Lieutenant 3               32009.72         
10 2020 Duty Officer               31947.58

3.4. Calculating Highest Retirement of Different Job sector of year 2020

In [56]:
sf_Retirement <- sf.data[c( 'Year','Job','Retirement')]
sf_Retirement <- sf_Retirement[sf_Retirement$Year == '2020',]
sf_Retirement <- sf_Retirement %>% group_by('Job') %>% arrange(desc(Retirement))%>% head(n=5)

sf_Retirement[4] <- NULL

sf_Retirement

Year Job                         Retirement
1 2020 Chief Investment Officer    114934.16 
2 2020 Chief Deputy Sheriff (PERS)  99840.50 
3 2020 Undersheriff                 97997.87 
4 2020 Managing Director            96118.05 
5 2020 Managing Director            96039.33

3.5. Calculating Highest Total Compensation of Different Job sector of year 2020

In [71]:
sf_Total.Compensation <- sf.data[c( 'Year','Job','Total.Compensation')]
sf_Total.Compensation <- sf_Total.Compensation %>% group_by('Job') %>% arrange(desc(Total.Compensation))
sf_Total.Compensation <- sf_Total.Compensation[sf_Total.Compensation$Year == "2020",]%>% head(n=10)
#sf_Total.Compensation%>% head(n=10)

sf_Total.Compensation[4] <- NULL

sf_Total.Compensation

Year Job                           Total.Compensation
1  2020 Chief Investment Officer      735942.8          
2  2020 Managing Director             627801.2          
3  2020 Managing Director             627506.6          
4  2020 Physician Administrator, DPH  595356.3          
5  2020 Managing Director             591936.5          
6  2020 Managing Director             580291.2          
7  2020 Physician Administrator, DPH  569142.4          
8  2020 Asst Med Examiner             567767.1          
9  2020 Senior Deputy Sheriff         554929.3          
10 2020 Sr Psychiatric Physician Spec 516599.6

#***4.Single Vector***

In [72]:
sf_Retirement <- sf.data[c( 'Year','Job','Retirement')]
sf_Retirement <- sf_Retirement[sf_Retirement$Year == '2020',]
sf_Retirement <- sf_Retirement %>% group_by('Job') %>% arrange(desc(Retirement))%>% head(n=10)

sf_Retirement[4] <- NULL

sf_Retirement

Year Job                          Retirement
1  2020 Chief Investment Officer     114934.16 
2  2020 Chief Deputy Sheriff (PERS)   99840.50 
3  2020 Undersheriff                  97997.87 
4  2020 Managing Director             96118.05 
5  2020 Managing Director             96039.33 
6  2020 Physician Administrator, DPH  94900.65 
7  2020 Sheriff                       94624.87 
8  2020 Managing Director             91763.58 
9  2020 Physician Administrator, DPH  90381.73 
10 2020 Managing Director             89844.26

In [73]:
vector.sf_Retirement <- c(sf_Retirement$Job)
head(vector.sf_Retirement ,5)


[1] "Chief Investment Officer"    "Chief Deputy Sheriff (PERS)"
[3] "Undersheriff"                "Managing Director"          
[5] "Managing Director"

The provided data appears to represent various job titles along with the corresponding retirement fund amounts for the year 2020. Here's a business explanation of this data:

In the year 2020, the organization (or government agency) had employees holding different job titles, and these employees were contributing to a retirement fund. The retirement fund is a financial arrangement where employees set aside a portion of their earnings during their working years to ensure financial security and stability after retirement.

1. **Chief Investment Officer**: The Chief Investment Officer in 2020 had a retirement fund contribution of $114,934.16. This indicates that the organization's top investment officer was actively contributing to their retirement savings, aligning with the importance of prudent financial planning even at the executive level.

2. **Chief Deputy Sheriff (PERS)**: The Chief Deputy Sheriff contributed $99,840.50 to the retirement fund in 2020. This suggests that even high-ranking law enforcement officials were committed to securing their financial futures through retirement planning.

3. **Undersheriff**: The Undersheriff, a key leadership position, contributed $97,997.87 to the retirement fund in 2020. This underscores the importance of retirement savings for leadership roles within the organization.

4. **Managing Director**: Managing Directors, possibly from different departments or divisions, had varying retirement fund contributions. This indicates that individuals in these roles were making decisions about their retirement savings based on their unique financial situations.

5. **Physician Administrator, DPH**: Employees with the title of Physician Administrator in the Department of Public Health (DPH) contributed to their retirement fund in 2020. This shows that healthcare professionals also prioritize retirement planning.

6. **Sheriff**: The Sheriff, a prominent law enforcement position, contributed $94,624.87 to the retirement fund. This demonstrates the commitment of top law enforcement officers to long-term financial security.

Overall, this data reflects a responsible approach to financial planning and retirement savings within the organization. Employees from various job roles and levels of responsibility are actively contributing to their retirement funds, acknowledging the importance of financial preparedness for life after their careers. It also indicates that the organization may offer retirement benefits or pension plans to its employees to facilitate their retirement savings.

#***5.LISTS***


5.1. Highest Retirement  for the Year 2020

In [215]:
# Extract the column as a vector

sf_Retirementvector <- sf_Retirement$Job


# Convert the vector to a list
sf_Retirementlist <- as.list(sf_Retirementvector)

sf_Retirementlist<- unique(sf_Retirementlist)
cat("- This list consists of Organizations Groups that offered Highest Retirement  for the Year 2020")



# Display the result
(sf_Retirementlist)

- This list consists of Organizations Groups that offered Highest Retirement  for the Year 2020

[[1]]
[1] "Chief Investment Officer"

[[2]]
[1] "Chief Deputy Sheriff (PERS)"

[[3]]
[1] "Undersheriff"

[[4]]
[1] "Managing Director"

[[5]]
[1] "Physician Administrator, DPH"

[[6]]
[1] "Sheriff"

Certainly, here's a brief explanation for each of the listed organization groups that offered the highest retirement benefits in the year 2020:

1. **Chief Investment Officer**: This role typically involves managing investment portfolios, and the high retirement benefits may reflect the importance of retaining experienced professionals in this critical financial position.

2. **Chief Deputy Sheriff (PERS)**: As a high-ranking law enforcement role, Chief Deputy Sheriffs often receive substantial retirement benefits to recognize their years of service and the demanding nature of their job.

3. **Undersheriff**: Similar to Chief Deputy Sheriffs, Undersheriffs play key roles in law enforcement agencies, which may result in higher retirement benefits due to their leadership responsibilities.

4. **Managing Director**: Managing Directors are often top executives in organizations, and their high retirement benefits may signify the importance of retaining top talent in leadership positions.

5. **Physician Administrator, DPH**: Given the critical role of healthcare administrators, particularly in public health, these professionals may receive significant retirement benefits to reward their contributions to healthcare systems.

6. **Sheriff**: As the highest-ranking law enforcement official in many jurisdictions, Sheriffs may receive substantial retirement benefits in recognition of their leadership and public safety responsibilities.

These benefits are likely reflective of the seniority, responsibility, and importance of these positions within their respective organizations.

5.2. Highest overtime for the Year 2020

In [211]:
sf_overtimevector <- sf_overtime$Job


# Convert the vector to a list
sf_overtimelist <- as.list(sf_overtimevector)

sf_overtimelist<- unique(sf_overtimelist)
cat("- This list consists of Organizations Groups that offered highest overtime for the Year 2020")



# Display the result
(sf_overtimelist)

- This list consists of Organizations Groups that offered highest overtime for the Year 2020

[[1]]
[1] "Senior Deputy Sheriff"

[[2]]
[1] "Deputy Sheriff"

[[3]]
[1] "Sheriff's Lieutenant"

Certainly, here's a brief explanation of the overtime benefits for each of the listed roles in the year 2020:

1. **Senior Deputy Sheriff**: Senior Deputy Sheriffs likely accrued high overtime due to their seniority and the demanding nature of law enforcement, often involving longer hours for public safety and emergency response.

2. **Deputy Sheriff**: Deputies often work overtime to ensure law enforcement coverage around the clock, responding to emergencies, and maintaining public safety, which can lead to substantial overtime benefits.

3. **Sheriff's Lieutenant**: Sheriff's Lieutenants, as supervisory roles, may have taken on additional responsibilities during critical situations, resulting in higher overtime benefits to manage law enforcement operations effectively.

These roles' overtime benefits reflect the crucial and often unpredictable nature of law enforcement work, requiring extended hours to ensure public safety and effective response to various situations.

5.3. Highest Total Compensation for the Year 2020

In [212]:
# Extract the column as a vector

sf_Total.Compensationvector  <- sf_Total.Compensation


# Convert the vector to a list
sf_Total.Compensationlist <- as.list(sf_Total.Compensationvector)

sf_Total.Compensationlist<- unique(sf_Total.Compensationlist)
cat("- This list consists of Organizations Groups that offered highest Total Compensation for the Year 2020")





# Display the result
(sf_Total.Compensationlist)

- This list consists of Organizations Groups that offered highest Total Compensation for the Year 2020

[[1]]
[1] "Chief Investment Officer"

[[2]]
[1] "Managing Director"

[[3]]
[1] "Physician Administrator, DPH"

[[4]]
[1] "Asst Med Examiner"

[[5]]
[1] "Senior Deputy Sheriff"

[[6]]
[1] "Sr Psychiatric Physician Spec"

Certainly, here's a brief business explanation of the total compensation for each of the listed roles in the year 2020:

1. **Chief Investment Officer**: This role often involves managing significant financial portfolios, and the high total compensation reflects the responsibility and expertise required in the financial sector.

2. **Managing Director**: Managing Directors typically hold top leadership positions in organizations, justifying their high total compensation as a recognition of their strategic leadership and decision-making.

3. **Physician Administrator, DPH**: Given the vital role of healthcare administrators, especially in public health, their total compensation reflects their responsibilities in managing healthcare systems and ensuring public well-being.

4. **Asst Med Examiner**: Assistant Medical Examiners play critical roles in forensic medicine, and their total compensation reflects their expertise in handling complex cases related to cause of death and forensic examinations.

5. **Senior Deputy Sheriff**: Senior Deputy Sheriffs receive high total compensation due to their seniority and the demanding nature of law enforcement roles, involving leadership and public safety responsibilities.

6. **Sr Psychiatric Physician Spec**: Senior Psychiatric Physician Specialists often deal with complex mental health cases, and their total compensation reflects their expertise in providing specialized psychiatric care.

These roles' total compensation packages are typically determined by factors such as their level of responsibility, expertise, and the critical nature of their contributions to their respective fields.

5.4. Highest Salary for the Year 2020

In [213]:
sf_salary_1vector <- sf_salary_1$Job


# Convert the vector to a list
sf_salary_1list <- as.list(sf_salary_1vector)

sf_salary_1list<- unique(sf_salary_1list)
cat("- This list consists of Organizations Groups that offered highest salary for the Year 2020")



# Display the result
(sf_salary_1list)

- This list consists of Organizations Groups that offered highest salary for the Year 2020

[[1]]
[1] "Chief Investment Officer"

[[2]]
[1] "Managing Director"

[[3]]
[1] "Physician Administrator, DPH"

[[4]]
[1] "Director"

[[5]]
[1] "Executive Contract Employee"

Certainly, here's a brief explanation of each term related to the job role that received the highest salary in 2020:

1. **Chief Investment Officer**: As a top financial executive, Chief Investment Officers oversee investment strategies, managing substantial portfolios, and their high salary reflects their responsibility for maximizing returns on investments.

2. **Managing Director**: Managing Directors typically hold senior leadership positions in organizations, responsible for strategic decision-making, and their high salary acknowledges their crucial role in shaping the company's direction.

3. **Physician Administrator, DPH**: Physician Administrators in public health manage healthcare systems, ensuring efficient operations, and their high salary recognizes their role in maintaining public well-being and healthcare quality.

4. **Director**: Directors often hold leadership roles overseeing specific departments or functions within organizations, and their high salary reflects their responsibility for guiding and achieving departmental goals.

5. **Executive Contract Employee**: Executive Contract Employees may be hired for specialized roles on a contract basis, often bringing unique expertise, and their high salary can result from the specialized skills and experience they offer.

The job role receiving the highest salary likely combines extensive responsibilities, expertise, and leadership in their field or organization, warranting a substantial compensation package.

5.5. Highest Health and Dental Benefit for the Year 2020

In [216]:
sf_healthvector <- sf_health$Job


# Convert the vector to a list
sf_healthlist <- as.list(sf_healthvector)

sf_healthlist<- unique(sf_healthlist)
cat("- This list consists of Organizations Groups that offered highest health and dental benefit for the Year 2020")



# Display the result
(sf_healthlist)

- This list consists of Organizations Groups that offered highest health and dental benefit for the Year 2020

[[1]]
[1] "Manager VI"

[[2]]
[1] "Manager III"

[[3]]
[1] "Court Manager"

[[4]]
[1] "Court Computer Sys Manager"

[[5]]
[1] "Manager I"

[[6]]
[1] "Manager II, MTA"

[[7]]
[1] "Manager V"

[[8]]
[1] "Lieutenant 3"

[[9]]
[1] "Duty Officer"

Certainly, here's a brief explanation of each of the job roles that received the highest health and dental benefits in 2020:

1. **Manager VI**: Managers at this level often oversee significant departments or divisions, and their high health and dental benefits reflect the value placed on retaining experienced leadership.

2. **Manager III**: Manager III roles typically involve middle-level management, and their generous health and dental benefits may aim to attract and retain skilled managers who play vital roles in organizational operations.

3. **Court Manager**: Court Managers are responsible for the efficient operation of court systems, and their substantial health and dental benefits acknowledge their role in ensuring justice administration.

4. **Court Computer Sys Manager**: Court Computer System Managers oversee technology systems within the court, and their high benefits recognize their critical role in maintaining court technology infrastructure.

5. **Manager I**: Manager I roles often represent entry-level management positions, and their health and dental benefits aim to support the development of future leaders within the organization.

6. **Manager II, MTA**: Managers in the transportation sector, like Manager II at MTA, may receive significant health and dental benefits due to the importance of transportation services in urban areas.

7. **Manager V**: Manager V positions typically entail significant responsibilities, and the high health and dental benefits may be a way to attract and retain top-level management talent.

8. **Lieutenant 3**: Lieutenants in law enforcement are often senior officers with supervisory roles, and their substantial health and dental benefits reflect their experience and responsibilities in maintaining public safety.

9. **Duty Officer**: Duty Officers play crucial roles in emergency response and management, and their high benefits may be a recognition of their contributions to public safety and crisis management.

These roles' generous health and dental benefits reflect their importance in various sectors and their impact on organizational effectiveness and public service.

#***6.Conditions***

6.1. Highest Retirement of Various Job role of Year 2020

In [165]:
sf_Retirement <- sf.data[c( 'Year','Job','Retirement')]
sf_Retirement <- sf_Retirement[sf_Retirement$Year == '2020',]
sf_Retirement <- sf_Retirement %>% group_by('Job') %>% arrange(desc(Retirement))%>% head(n=10)
sf_Retirement <-  sf_Retirement %>% filter(Retirement > 50000)
sf_Retirement[4] <- NULL
dim(sf_Retirement)
cat("These Shows Number of Rows")
sf_Retirement

[1] 10  3

These Shows Number of Rows

Year Job                          Retirement
1  2020 Chief Investment Officer     114934.16 
2  2020 Chief Deputy Sheriff (PERS)   99840.50 
3  2020 Undersheriff                  97997.87 
4  2020 Managing Director             96118.05 
5  2020 Managing Director             96039.33 
6  2020 Physician Administrator, DPH  94900.65 
7  2020 Sheriff                       94624.87 
8  2020 Managing Director             91763.58 
9  2020 Physician Administrator, DPH  90381.73 
10 2020 Managing Director             89844.26

In 2020, the Chief Investment Officer received a retirement benefit of $114,934.16, reflecting the financial expertise and leadership role within the organization. Managing Directors and the Physician Administrator, DPH, also received substantial retirement benefits, underscoring the value placed on their strategic leadership and contributions to healthcare and organizational success.

6.2. Highest Overtime of Various Job role of Year 2020


In [150]:
sf_overtime <- sf.data[c( 'Year','Job','Overtime')]
sf_overtime <- sf_overtime [sf_overtime$Year == '2020',]
sf_overtime <- sf_overtime %>% group_by('Job') %>% arrange(desc(Overtime))%>% head(n=10)
sf_overtime <- sf_overtime %>% filter(Overtime > 100000)
sf_overtime[4] <- NULL
sf_overtime

Year Job                   Overtime
1  2020 Senior Deputy Sheriff 317589.4
2  2020 Deputy Sheriff        272856.4
3  2020 Deputy Sheriff        262417.2
4  2020 Senior Deputy Sheriff 249774.1
5  2020 Senior Deputy Sheriff 238600.4
6  2020 Sheriff's Lieutenant  214921.0
7  2020 Sheriff's Lieutenant  214900.2
8  2020 Deputy Sheriff        213483.0
9  2020 Senior Deputy Sheriff 213198.3
10 2020 Sheriff's Lieutenant  209889.9

In 2020, Senior Deputy Sheriffs received the highest overtime earnings at $317,589.4, followed by Deputy Sheriffs and Sheriff's Lieutenants, indicating the additional compensation for their roles in law enforcement, often involving demanding and unpredictable work hours. These overtime benefits recognize the dedication and effort put into maintaining public safety.

6.3. Highest Health and Dental Benefits of Various Job role of Year 2020

In [154]:
sf_health <- sf.data[c( 'Year','Job','Health.and.Dental')]
sf_health <- sf_health [sf_health$Year == '2020',]
sf_health <- sf_health %>% group_by('Job') %>% arrange(desc(Health.and.Dental))%>% head(n=10)
sf_health <- sf_health %>% filter(Health.and.Dental > 7500)
sf_health  [4] <- NULL
sf_health

Year Job                        Health.and.Dental
1  2020 Manager VI                 49000.22         
2  2020 Manager III                43446.48         
3  2020 Court Manager              35633.00         
4  2020 Court Computer Sys Manager 35633.00         
5  2020 Manager I                  33480.25         
6  2020 Manager II, MTA            33479.10         
7  2020 Manager V                  32363.01         
8  2020 Lieutenant 3               32257.85         
9  2020 Lieutenant 3               32009.72         
10 2020 Duty Officer               31947.58

In 2020, the job roles of Manager VI, Manager III, and Court Manager received the highest health and dental benefits, totaling $49,000.22, $43,446.48, and $35,633.00, respectively. These substantial benefits reflect the organization's commitment to ensuring the well-being and healthcare coverage of its managerial staff. Additionally, Lieutenant 3 and Duty Officer positions also received significant health and dental benefits, highlighting the importance of employee health in their roles within the organization.

#***7.loops***

#7.1.Highest Salary

In [186]:
# Create an empty data frame to store the results
highest_salary_and_job_family_by_year <- data.frame(Year = integer(), Highest_Total_Salary = numeric(), Highest_Job_Family = character())

# Get unique years in the dataset
unique_years <- unique(sf.data$Year)

# Loop through each unique year
for (year in unique_years) {
  # Filter the data for the current year
  year_data <- sf.data[sf.data$Year == year, ]

  # Find the highest Total.Salary for employees in this year
  highest_salary <- max(year_data$Total.Salary, na.rm = TRUE)

  # Find the corresponding Job.Family for the employee with the highest salary
  highest_salary_job_family <- year_data[year_data$Total.Salary == highest_salary, ]$Job.Family

  # Create a data frame with the year, highest Total.Salary, and highest Job.Family
  year_result <- data.frame(Year = year, Highest_Total_Salary = highest_salary, Highest_Job_Family = highest_salary_job_family)

  # Append the result to the final data frame
  highest_salary_and_job_family_by_year <- rbind(highest_salary_and_job_family_by_year, year_result)
}

# Print the resulting data frame
highest_salary_and_job_family_by_year


Year Highest_Total_Salary Highest_Job_Family           
1 2013 381716.8             Fire Services                
2 2014 372637.8             Police Services              
3 2015 409892.6             Med Therapy & Auxiliary      
4 2016 515101.8             Administrative & Mgmt (Unrep)
5 2017 533985.9             Administrative & Mgmt (Unrep)
6 2018 550065.9             Administrative & Mgmt (Unrep)
7 2019 658867.6             Police Services              
8 2020 584297.6             Administrative & Mgmt (Unrep)

From 2013 to 2020, the highest salaries were observed in various organizational groups. Notably, in 2019, the Police Services group had the highest salary at  658,867.6, reflecting the competitive compensation offered to support law enforcement efforts. In the following years, from 2016 to 2018, the Administrative & Management (Unrepresented) group exhibited substantial salaries, with a peak of 550,065.9 in 2018, indicating attractive remuneration for managerial and administrative roles. Additionally, 2017 saw a significant rise in salaries in this group, reaching $533,985.9. These fluctuations in highest salaries across different years highlight the dynamic nature of compensation within diverse organizational sectors.

#7.2.lowest Retirement

In [188]:
# Create an empty data frame to store the results
lowest_retirement_and_job_family_by_year <- data.frame(Year = integer(), Lowest_Retirement = numeric(), Lowest_Job_Family = character())

# Get unique years in the dataset
unique_years <- unique(sf.data$Year)

# Loop through each unique year
for (year in unique_years) {
  # Filter the data for the current year
  year_data <- sf.data[sf.data$Year == year, ]

  # Find the lowest Retirement for employees in this year
  lowest_retirement <- min(year_data$Retirement, na.rm = TRUE)

  # Find the corresponding Job.Family for the employee with the lowest retirement
  lowest_retirement_job_family <- year_data[year_data$Retirement == lowest_retirement, ]$Job.Family

  # Create a data frame with the year, lowest retirement, and lowest job family
  year_result <- data.frame(Year = year, Lowest_Retirement = lowest_retirement, Lowest_Job_Family = lowest_retirement_job_family)

  # Append the result to the final data frame
  lowest_retirement_and_job_family_by_year <- rbind(lowest_retirement_and_job_family_by_year, year_result)
}

# Print the resulting data frame
lowest_retirement_and_job_family_by_year


Year Lowest_Retirement Lowest_Job_Family
1 2013 -21302.02         Nursing          
2 2014 -28723.19         Nursing          
3 2015 -21137.23         Nursing          
4 2016  -1069.96         Skilled Labor    
5 2017  -6129.43         Legal & Court    
6 2017  -6129.43         Legal & Court    
7 2018 -26670.15         Nursing          
8 2019  -1372.34         Legal & Court    
9 2020  -1603.77         Personnel

Between 2013 and 2020, the lowest retirement benefits were observed in various job families. Notably, in 2016, the job family "Skilled Labor" had the lowest retirement benefits at 1,069.96, indicating a relatively limited retirement package for this category of workers. The year 2017 saw the "Legal & Court" job family with two instances of low retirement benefits, suggesting some inconsistency within this group. In 2019, another "Legal & Court" job family entry showed low retirement benefits at $1,372.34. These fluctuations in retirement benefits across different job families and years underscore the disparities in retirement planning and support within various employment sectors.

#7.3.Lowest Salaries

In [189]:
# Create an empty data frame to store the results
lowest_salaries_and_job_family_by_year <- data.frame(Year = integer(), Lowest_Salaries = numeric(), Lowest_Job_Family = character())

# Get unique years in the dataset
unique_years <- unique(sf.data$Year)

# Loop through each unique year
for (year in unique_years) {
  # Filter the data for the current year
  year_data <- sf.data[sf.data$Year == year, ]

  # Find the lowest Salaries for employees in this year
  lowest_salaries <- min(year_data$Salaries, na.rm = TRUE)

  # Find the corresponding Job.Family for the employee with the lowest Salaries
  lowest_salaries_job_family <- year_data[year_data$Salaries == lowest_salaries, ]$Job.Family

  # Create a data frame with the year, lowest Salaries, and lowest Job.Family
  year_result <- data.frame(Year = year, Lowest_Salaries = lowest_salaries, Lowest_Job_Family = lowest_salaries_job_family)

  # Append the result to the final data frame
  lowest_salaries_and_job_family_by_year <- rbind(lowest_salaries_and_job_family_by_year, year_result)
}

# Print the resulting data frame
lowest_salaries_and_job_family_by_year


Year Lowest_Salaries Lowest_Job_Family      
1 2013 -17635.32       Human Services         
2 2014 -68771.78       Management             
3 2015  -7423.00       Hospital Administration
4 2016  -4770.20       Skilled Labor          
5 2017 -33808.20       Legal & Court          
6 2017 -33808.20       Legal & Court          
7 2018 -15528.00       Management             
8 2019 -24605.54       Nursing                
9 2020 -10854.00       Management

From 2013 to 2019, the lowest salaries were found across different job families and departments. In 2016, the "Skilled Labor" category had the lowest salaries at 4,770.20, indicating relatively lower earnings within this job family. The year 2015 saw Hospital Administration with notably low salaries at 7,423.00. Additionally, 2017 showed two entries for the "Legal & Court" job family with low salaries at $33,808.20, suggesting potential disparities in compensation within this sector. These fluctuations in salaries across various job families and years reflect variations in income levels and compensation practices in different employment domains.

#***8.Summary***

Provided data showcases the organization's commitment to ensuring the financial security and well-being of its diverse workforce. It is evident that employees in crucial leadership positions, such as the Chief Investment Officer and Chief Deputy Sheriff (PERS), received substantial retirement benefits. This highlights the organization's recognition of the importance of providing comprehensive retirement planning and support to key personnel who hold critical roles within the organization. Even leadership positions like Undersheriff and Managing Director received significant retirement benefits, emphasizing the organization's dedication to helping its leaders secure their financial futures.

Moreover, the data underscores the organization's inclusive approach to retirement planning, as employees from various job titles and levels actively contributed to their retirement funds. Notably, healthcare professionals like the Physician Administrator in the Department of Public Health (DPH) also participated in retirement planning, indicating the importance of securing the financial well-being of those dedicated to healthcare. Additionally, the retirement benefits provided to positions like Sheriff highlight the organization's commitment to supporting public safety leaders in achieving long-term financial stability. In summary, this data reflects a responsible and equitable approach to retirement planning, showcasing the organization's unwavering dedication to ensuring the financial security of its entire workforce, from top executives to frontline professionals.

#***9. Recommendations***

Based on the insights derived from the provided data, several recommendations can be made to optimize the organization's workforce management and compensation strategies.

Firstly, the organization should consider implementing a structured retirement planning and financial literacy program for all employees, irrespective of their job roles. This would empower employees with the knowledge and tools necessary to make informed decisions about their retirement savings. Additionally, offering a retirement fund matching program could incentivize employees to actively contribute to their retirement accounts, thereby strengthening their long-term financial security.

Secondly, it's essential for the organization to conduct regular salary and compensation benchmarking to ensure that its compensation packages remain competitive within the industry. This would help attract and retain top talent across various job families and departments. Furthermore, addressing disparities in compensation observed in certain job families, such as "Skilled Labor" and "Legal & Court," should be a priority to ensure fairness and equity across all roles.

Overall, a comprehensive approach to employee financial well-being, including retirement planning and competitive compensation practices, will not only enhance the organization's ability to attract and retain top talent but also demonstrate its commitment to supporting employees in their career development and long-term financial goals.